### ISSUE TITLE
Target Selection at the milling angle

### ISSUE THREAD

**Issue by hvdrosa** [eisfabian/PACEtomo#22]
Hi Fabian,

I'm a PhD student in Simone Mattei's team at the EMBL Imaging Centre. 

Zhengyi and I have made a change to our workflow, and we would like to discuss it with you: 

We have commented out the line `sem.TiltTo(0)` in the selectTargets script to be able to select targets at the milling angle, taking the pre-tilt into account. This change has slightly improved the image quality for reference images, as the beam is now perpendicular to the lamella and the cross-section/thickness is slightly smaller. However, we have concerns about potential issues in the image-shift calculations performed by your scripts, which seem to assume a 0-degree stage tilt for reference acquisition.

Our understanding is that a 10-degree pre-tilt can result in an image shift difference of approximately 1.5 um for a 10 um image shift. This raises uncertainty about possible future problems for larger shifts, although we haven't encountered any issues with target selection and acquisitions so far using this method.

We do have noticed that executing the "measure geometry" script at the milling angle reduces its precision (which is expected), and the output "pre-tilt" angle only reflects the difference from the current angle, not from a 0-degree stage tilt. Nevertheless, we are exploring ways to optimize our workflow. One possibility is to initially try to measure geometry at a 0-degree stage tilt to determine the pre-tilt and then acquire the Modified Milled Maps (MMMs) and select targets at the pre-tilt angle. Additionally, Zhengyi is experimenting with using cropped regions from these pre-tilted MMMs, as done before with your collaboration.

So, in summary, we have two questions:

1-Is there a specific reason for maintaining all operations at 0 degrees, disregarding the milling angle? Or is it only for user convenience, as you only need to collect MMMs at 0 degrees without caring about the pre-tilt before?

2-Can you identify any potential issues or drawbacks with our current approach that we might have overlooked?

We look forward to your comments on this. :)
Thanks a lot for all the help!

All the Best,
Higor

**Comment by eisfabian** [comment]
Hi Higor,

Thanks for reaching out!
I understand what you want to do. You are right, collecting the target images at the same startTilt might slightly improve the correlation of the realignment process.
However, PACEtomo currently does not support target selection at non-zero stage tilts and it did not cause me any issues so far.

I could include the conversion of the tilted image shifts to the untilted image shifts during target selection, but that would indeed require the pretilt and rotation parameters. And this is the answer to your question 1: The reason for selecting targets at an untilted stage was because of the unknown sample geometry.

I would have to either force users to enter the parameters/run the measureGeometry routine before picking targets, which is suboptimal or I need to postpone the saving of the targets file until the whole procedure finishes. Thus far I was saving the targets as you chose them so you could just stop the script whenever you want and still have a valid target selection.

(2) As you mentioned, there will be an error in the image shifts, which might not be an issue if your field of view is large enough and your Y image shifts are small enough.
Regarding the "measure geometry" routine, you could just close the target selection, tilt the stage to 0, rerun selectTargets on the same target file, say you want to edit the target file and then run the "measure geometry" routine to get the proper values.

I will think about it some more, because it will be a compromise either way.
Do you have any direct comparison between the two approaches? If there is a measurable improvement in target realignment, it might be worth the tradeoff.

Thanks again and best wishes,
Fabian
